src/hooks/useTokenBalance.ts

```diff
@@ -7,25 +7,42 @@ import useWeb3 from './useWeb3'
 import useRefresh from './useRefresh'
 import useLastUpdated from './useLastUpdated'
 
+export type FetchStatus = 'notFetched' | 'fetching' | 'success' | 'failed'
```

Comment:
A gentle reminder - what I said about this pattern was just a _suggestion_. :)

(and, recently I have been using enums)

Reply:
I liked it! I think introducing these patterns is pretty light. I stripped out the 'fetching' state - as it has no explicit use atm and effectively re-triggers a 'loading' state every 6s for the users.

Enums ✅ 
Typed state ✅